metals/src/main/scala/scala/meta/internal/metals/FoldingMode.scala

```diff
@@ -0,0 +1,22 @@
+package scala.meta.internal.metals
+
+import org.eclipse.lsp4j.FoldingRange
+
+trait FoldingMode {
```

Comment:
Can we please inline this into the `FoldingRangeProvider` file? I think it's nice to have everything related to folding in the same file (as long as it doesn't grow huge).